CMakeLists.txt

```diff
@@ -50,13 +50,21 @@ include(pybind11)
 include(GoogleBenchmark)
 include(GenerateExportHeader)
 find_package(Boost 1.67 REQUIRED)
+find_package(TBB CONFIG)
+if(TBB_FOUND)
+  configure_file(core/include/scipp/core/parallel-tbb.h core/include/scipp/core/parallel.h COPYONLY)
+else()
+  configure_file(core/include/scipp/core/parallel-fallback.h core/include/scipp/core/parallel.h COPYONLY)
```

Comment:
It might also be useful to deliberately compile with this for measurement or debugging purposes. We might consider an additional cmake flag in future to support that.